src/app/ReadClient.cpp

```diff
@@ -757,6 +802,24 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
                 attributePath.mListOp = ConcreteDataAttributePath::ListOperation::ReplaceAll;
             }
 
+            if (attributePath ==
+                ConcreteDataAttributePath(kRootEndpointId, Clusters::IcdManagement::Id,
+                                          Clusters::IcdManagement::Attributes::OperatingMode::Id))
+            {
+                PeerType peerType;
+                if (CHIP_NO_ERROR == ReadICDOperationModeFromAttributeDataIB(dataReader, peerType))
+                {
+                    // Since we are in the middle of parsing the attribute data, it is safe to call `OnPeerTypeChange`
+                    // As it will only update the LIT / SIT bit for us without triggerring resubscription on this read client.
```

Comment:
Why won't it trigger resubscription on this read client?  The key is that we don't think we need resubscription, so OnActiveModeNotification is a no-op, right?  Please say that.

Reply:
Updated.

I also noticed that there might be a corner case that we might in InactiveSubscription state but an unexpected report is received. Also updated the `ReadClient::OnMessageReceived` to ensure this will never happen.